# HG changeset patch
# User Carlo de Falco <kingcrimson@tiscali.it>
# Date 1330897961 -3600
#      Sun Mar 04 22:52:41 2012 +0100
# Node ID df5488e46dca4559cfbd9b4808cfca14cf412c7a
# Parent  eff4a5933e28539ebcb120d0f42c715d94e7511f
fix bug in input validation for textread

* textread.m: fix a bug in input validation when
headerlines is the only option passed

diff --git a/scripts/io/textread.m b/scripts/io/textread.m
--- a/scripts/io/textread.m
+++ b/scripts/io/textread.m
@@ -66,18 +66,18 @@ function varargout = textread (filename,
 
   ## Skip header lines if requested
   headerlines = find (strcmpi (varargin, "headerlines"), 1);
   ## Beware of zero valued headerline, fskipl would skip to EOF
   if (! isempty (headerlines) && (varargin{headerlines + 1} > 0))
     fskipl (fid, varargin{headerlines + 1});
     varargin(headerlines:headerlines+1) = [];
   endif
-
-  if (nargin > 2 && isnumeric (varargin{1}))
+  
+  if (! isempty (varargin) && isnumeric (varargin{1}))
     nlines = varargin{1};
   else
     nlines = Inf;
   endif
 
   if (isfinite (nlines) && (nlines >= 0))
     str = tmp_str = "";
     n = 0;
@@ -135,14 +135,22 @@ endfunction
 %! d = rand (5, 3);
 %! dlmwrite (f, d, "precision", "%5.2f");
 %! [a, b, c] = textread (f, "%f %f %f", "delimiter", ",", "headerlines", 3);
 %! unlink (f);
 %! assert (a, d(4:5, 1), 1e-2);
 %! assert (b, d(4:5, 2), 1e-2);
 %! assert (c, d(4:5, 3), 1e-2);
 
+%!test
+%! f = tmpnam ();
+%! d = rand (7, 2);
+%! dlmwrite (f, d, "precision", "%5.2f");
+%! [a, b] = textread (f, "%f, %f", "headerlines", 1);
+%! unlink (f);
+%! assert (a, d(2:7, 1), 1e-2);
+
 %% Test input validation
 %!error textread ()
 %!error textread (1)
 %!error <arguments must be strings> textread (1, "%f")
 %!error <arguments must be strings> textread ("fname", 1)
 
